### ISSUE TITLE
more generic approach; another use case

### ISSUE THREAD

**Issue by ghost** [someshsingh22/FlaiReddit-MIDAS#1]
Hi,

Hope you are all well !

I just wanted to ask you if your approach could apply to classify arxiv abstracts into specific categories/tasks ?

More precisely, I have a dataset of 90k abstracts classified in tasks.
https://paper2code.com/public/tag_predict.txt.tar.gz

How can I train a model based on this dataset ?

Cheers,
X

**Comment by someshsingh22** [comment]
Hi @x0rzkov,
Yes, you can extend this approach to any text classification task/dataset.
For replicating these results you need two things:

1. Extracting data from a given URL, i.e. given any particular URL extract its data (id, publisher, title, summary, tasks in your case) as used in https://github.com/someshsingh22/FlaiReddit-MIDAS/blob/4105bd69ed9da2542bc902cd2af21b3dadb593c4/app/main.py#L25

You can check the implementation at https://github.com/someshsingh22/FlaiReddit-MIDAS/blob/4105bd69ed9da2542bc902cd2af21b3dadb593c4/app/tfidf_model.py#L14

2. A model that maps your data to the given category, in this repository I have simply pickled and stored my [model](https://github.com/someshsingh22/FlaiReddit-MIDAS/blob/master/app/model.pickle)

You can check the implementation (This is done separately to avoid unnecessary load on the server)
Note that I have not used the model mentioned [here](https://github.com/someshsingh22/FlaiReddit-MIDAS/blob/master/Notebooks/Part-3-RedditBERT.ipynb) because it is a huge model and Heroku's free version won't allow that, instead used this smaller but effective [model](https://github.com/someshsingh22/FlaiReddit-MIDAS/blob/master/Notebooks/Part-2-EDA.ipynb).

If you clarify your exact issue it will be easier to explain.

**Comment by ghost** [comment]
One of my issue is that I would like to create prediction classifier as a service (like https://github.com/KeremZaman/semantic-sh) with a flask server.

What are the scripts to run from the cli to launch the training without notebooks ?

Ps. It would be awesome to dockerize it. I can make a PR for that.

**Comment by someshsingh22** [comment]
Great, I will add python scripts for creating model.pickle and train it with given data. #2 Will solve this

dockerizing it would be awesome! You can proceed with that

**Comment by ghost** [comment]
Is it supporting gpu ? asking for the dockerization.

**Comment by someshsingh22** [comment]
There are two models as I have mentioned,

- BERT - High Performance, Not Deployed, supports GPU (recommended due to number of parameters)

- Multinomial Naive Bayes - Low Performance, Deployed, does not and cannot support GPU (Again Heroku Free does not support GPU which is why I have deployed MNB)

**Comment by ghost** [comment]
So I ll prepare both cpu/gpu version.

ps.. It is my second handle on github.

**Comment by someshsingh22** [comment]
Yes, you can start with CPU because it takes only a few seconds to train and test it

**Comment by someshsingh22** [comment]
@lucmichalski I have implemented MNB (currently deployed) in #3 and given the instructions as well, you only have to change minimal variables depending upon your usage. I will add BERT in a further PR, feel free to close this if it works for you and create a separate issue for dockerizing.
